common/rdr/Exception.cxx

```diff
@@ -54,33 +54,16 @@ SystemException::SystemException(const char* s, int err_)
 {
   strncat(str_, ": ", len-1-strlen(str_));
 #ifdef _WIN32
-  // Windows error messages are crap, so use unix ones for common errors.
-  const char* msg = 0;
-  switch (err) {
-  case WSAECONNREFUSED: msg = "Connection refused";       break;
-  case WSAETIMEDOUT:    msg = "Connection timed out";     break;
-  case WSAECONNRESET:   msg = "Connection reset by peer"; break;
-  case WSAECONNABORTED: msg = "Connection aborted";       break;
-  }
-  if (msg) {
-    strncat(str_, msg, len-1-strlen(str_));
-  } else {
-#ifdef UNICODE
-    WCHAR* tmsg = new WCHAR[len-strlen(str_)];
-    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
-                  0, err, 0, tmsg, len-1-strlen(str_), 0);
-    WideCharToMultiByte(CP_ACP, 0, tmsg, wcslen(tmsg)+1,
-		              str_+strlen(str_), len-strlen(str_), 0, 0);
-    delete [] tmsg;
-#else
-    char* currStr = str_+strlen(str_);
-    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
-                  0, err, 0, currStr, len-1-strlen(str_), 0);
-#endif
-    int l = strlen(str_);
-    if ((l >= 2) && (str_[l-2] == '\r') && (str_[l-1] == '\n'))
-      str_[l-2] = 0;
-  }
+  wchar_t *currStr = new wchar_t[len-strlen(str_)];
+  FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                 0, err, 0, currStr, len-1-strlen(str_), 0);
+  WideCharToMultiByte(CP_UTF8, 0, currStr, wcslen(currStr), str_+strlen(str_),
+                      len-1-strlen(str_), 0, 0);
+  delete [] currStr;
+
+  // strtok() will split str_ once a delimiter is found, thus removing
+  // trailing characters.
+  strtok(str_, "\r");
```

Comment:
A bit of an abuse of `strtok()`. :)

I think the original code was quite sufficient in this case. Or use `strchr()` if you're paranoid about multiple line breaks.

Is this even an issue these days? Have you verified that this code is really needed?

Reply:
Unfortunately, the code is still needed. Connect to a non existent server through command line or GUI will result in different error outputs. The original code doesn't seem to be able to handle it correctly.

Reply:
Can you show how it looks in that case?

Reply:
This is how it looks with the original code when using GUI:

![error](https://user-images.githubusercontent.com/59053320/76518529-8d173300-645f-11ea-9911-78d4fe04cec5.png)